crates/sui-replay-frontend/src/lib.rs

```diff
@@ -0,0 +1,19 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use std::path::PathBuf;
+use std::{fs::File, io::Write};
+
+pub fn generate_html_from_json(data: &str, output_path: PathBuf, network: &str) {
+    let styles = include_str!("../frontend/sources/styles.css");
+    let file = include_str!("../frontend/sources/index.html");
+
+    let fixed_html = file
+        .replace("CLIENT_ENV_NETWORK", network)
+        .replace("REPLACE_ME", data)
+        .replace(".INSERT_STYLES_HERE{color:sui}", styles);
+
+    let mut file = File::create(output_path.clone()).ok().unwrap();
+    let _ = file.write_all(fixed_html.as_bytes());
+    open::that(output_path).ok();
```

Comment:
Could this be a bit more error proof? E.g., no unwrap, and check if the file was actually created and otherwise return the error. Same with write_all.